### PR TITLE
contrib: optimize TradingChartHeader react rendering perf.

### PR DESCRIPTION
**Pull request by 0xcryptovenom**
- Do not use `useEffect` unncessarily to perform redux / local state update synchronization
  - Instead, do the local state updates **when strictly necessary with appropriate safeguards** directly in the render.
    - In doing so, the rendering will be interrupted, local state updated immediately & a new render with the updated state will be triggered
- Subscribe to the current token's streamed prices from the Redux store, instead of all tokens.
- This change needs to be tested for possible regressions.

### CODE REVIEW

File: src/components/pages/trading/TradingChartHeader/TradingChartHeader.tsx [hunk 1]
```diff
@@ -24,45 +24,44 @@ export default function TradingChartHeader({
...
+      // if streamingTokenPrices is smaller than previous value, set color to red
+      const newTokenColor =
+        selectedTokenPrice > previousTokenPrice
+          ? 'text-green'
+          : selectedTokenPrice < previousTokenPrice
+          ? 'text-red'
+          : tokenColor;
+      // make sure we're only updating the local state if the new color is different.
+      if (newTokenColor !== tokenColor) {
+        setTokenColor('text-green');
```
Comment (0xcryptovenom) [inline src/components/pages/trading/TradingChartHeader/TradingChartHeader.tsx:50]: Doesn't work as expected, needs to be revisited.

Comment (0xcryptovenom) [reply src/components/pages/trading/TradingChartHeader/TradingChartHeader.tsx:50]: I'm an idiot lmao, fixing

File: src/components/pages/trading/TradingChartHeader/TradingChartHeader.tsx [hunk 2]
```diff
@@ -24,45 +24,44 @@ export default function TradingChartHeader({
   selected: Token;
   onChange: (t: Token) => void;
 }) {
-  const streamingTokenPrices = useSelector((s) => s.streamingTokenPrices);
+  const selectedTokenPrice = useSelector(
+    (s) => s.streamingTokenPrices[getTokenSymbol(selected.symbol)] ?? null,
+  );
   const stats = useDailyStats();
```
Comment (0xcryptovenom) [inline src/components/pages/trading/TradingChartHeader/TradingChartHeader.tsx:30]: Possibly broke daily stats, need to double-check

Comment (0xcryptovenom) [reply src/components/pages/trading/TradingChartHeader/TradingChartHeader.tsx:30]: No it works fine but there must be a race condition somewhere, need to look at possibly making it more resilient.

**Comment by adrena-orex** [comment]
Seems now that the colors changes way less often than previously in header.

Screenshot 2024-11-02 at 11 00 15

**Comment by 0xcryptovenom** [comment]
@adrena-orex I think you might have checked the previous deploy where I had made a mistake, I force-pushed since then.

If we redeploy we should have the same behavior as previously, just faster.

**Pull request merged**